File: src/main/java/com/skuniv/bigdata/domain/dto/open_api/BargainBodyDto.java
package com.skuniv.bigdata.domain.dto.open_api;

import lombok.Data;

import javax.xml.bind.annotation.*;
import java.util.List;

@Data
@XmlRootElement(name = "body")
@XmlAccessorType(XmlAccessType.FIELD)
public class BargainBodyDto {
    @XmlElementWrapper(name = "items")
    @XmlElement(name = "item")
    private List<BargainItemDto> item;

    @XmlElement(name = "numOfRows")
    private int numOfRows;

    @XmlElement(name = "pageNo")
    private int pageNo;

    @XmlElement(name = "totalCount")
    private int totalCount;
}

File: src/main/java/com/skuniv/bigdata/domain/dto/open_api/BargainBodyDto.kt
package com.skuniv.bigdata.domain.dto.open_api

import javax.xml.bind.annotation.*

@XmlRootElement(name = "body")
@XmlAccessorType(XmlAccessType.FIELD)
data class BargainBodyDto(

        @XmlElementWrapper(name = "items")
        @XmlElement(name = "item")
        var item: List<BargainItemDto>? = null,

        @XmlElement(name = "numOfRows")
        var numOfRows: Int = 0,

        @XmlElement(name = "pageNo")
        var pageNo: Int = 0,

        @XmlElement(name = "totalCount")
        var totalCount: Int = 0
)